packages/components/addon/components/hds/copy/button/index.hbs

```diff
@@ -0,0 +1,20 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: MPL-2.0
+}}
+<button
+  type="button"
+  class="{{this.classNames}} {{if this.isSuccess 'is-success'}}"
+  {{on "click" this.copyCode}}
+  ...attributes
+  aria-label={{if this.isIconOnly this.text null}}
+>
+  {{#if this.isIconOnly}}
+    <FlightIcon @name="{{if this.isSuccess 'clipboard-checked' 'clipboard-copy'}}" class="hds-copy-button__icon" />
+  {{else}}
+    <div class="hds-copy-button__text hds-typography-body-100 hds-font-weight-semibold">
+      {{this.text}}
+    </div>
+    <FlightIcon @name="{{if this.isSuccess 'clipboard-checked' 'clipboard-copy'}}" class="hds-copy-button__icon" />
+  {{/if}}
```

Comment:
I would use conditionals just around the div with the text to avoid repeating the code for the `FlightIcon`.

Reply:
Resolving, have refactored draft.